### DISCUSSION TITLE
Should notebooks work out of the box?

### DISCUSSION THREAD

**Discussion by mccalluc** [opendp/opendp#898] [Getting Involved] [answered]
I'm just taking a first look at this project, and I hit some bumps getting started. I didn't want to clutter your bug queue with this, since it is kind of amorphous. Some examples:

- The first cell of `first-look-at-DP.ipynb` fails because pandas is not installed. Could image adding a `!pip install pandas` in the cell, or adding pandas to `.binder/requirements.txt`.
- If someone tries to checkout the project and run jupyter locally, they may have the last release of opendp installed, but may be looking at notebook code updated for the next release. (or at least that's what I did.) Could imagine either changing the branch convention, so main is always the last release, or making it explicit in the notebook that the version in local checkout should be installed, and not the release on pypi.
- On smartnoise-sdk, the docs also link to Colab, but Colab has a lot of packages preinstalled, and those versions conflict with those required for the project. Either a pip uninstall to clean up the environment, or stop explicitly linking to Colab?

There are different ways these could be fixed, and the status quo is not too bad either.

**Comment by Shoeboxam** [comment, marked as answer]
They should work out of the box, but indeed the binder requirements are out of date. Thanks for pointing this out. I'll look into keeping the binder dep hard-linked to the relevant OpenDP release in future versions, and adjusting the links to point to a tag.

Feel free to open issues! And don't hesitate to open more for any other bumps you hit. This is valuable information for on-boarding.

**Reply by mccalluc** [reply]
Thanks: I've filed #903, and may try to make a PR.
